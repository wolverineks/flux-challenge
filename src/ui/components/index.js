// @flow

export {Layout, Header, Body} from './Layout'
export {ScrollableList, List, Item, ScrollButtons, ScrollButton} from './ScrollableList'
export {default as PlanetIndicator} from './PlanetIndicator'
export {default as Sith} from './Sith'
