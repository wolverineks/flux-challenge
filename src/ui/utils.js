// @flow

import type {Id, Siths} from '../state/types'

const prop = (key: string) => (object: ?Object) =>
  object
    ? object[key]
    : null

export const name = prop('name')
export const homeworld = prop('homeworld')

export const masters = (siths: Siths) => (id: Id) => {
  const masterId = siths[id].masterId
  if (!masterId) return [id]
  return [...masters(siths)(masterId), id]
}

export const apprentices = (siths: Siths) => (id: Id) => {
  const apprenticeId = siths[id].apprenticeId
  if (!apprenticeId) return [id]
  return [id, ...apprentices(siths)(apprenticeId)]
}
