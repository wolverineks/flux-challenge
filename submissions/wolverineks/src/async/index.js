// @flow

import {combinePixies} from 'redux-pixies'

import sithPixie from './sithPixie'
import sithsPixie from './sithsPixie'

export {
  sithPixie,
  sithsPixie
}

export default combinePixies({sithsPixie})
