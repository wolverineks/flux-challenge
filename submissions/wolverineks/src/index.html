<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <link rel="stylesheet" href="./styles.css">
  <title>Flux Challenge</title>
</head>

<body>
  <div id="root"></div>
</body>

</html>
