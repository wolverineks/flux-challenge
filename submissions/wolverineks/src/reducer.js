// reducer.js
// @flow

import {combineReducers} from 'redux'

import {reducer} from './SithDashboard'

export default combineReducers({
  sithDashboard: reducer
})
