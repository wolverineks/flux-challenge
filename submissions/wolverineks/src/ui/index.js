// @flow

import type {Children} from '../state/types'
import {Layout, Header, Body, PlanetIndicator, Sith, ScrollableList, ScrollButtons, ScrollButton, List, Item} from './components'
import classNames from './styles'

export type {Children}
export {
  Layout,
  Header,
  Body,
  PlanetIndicator,
  Sith,
  ScrollableList,
  ScrollButtons,
  ScrollButton,
  List,
  Item,
  classNames
}
export default Layout
