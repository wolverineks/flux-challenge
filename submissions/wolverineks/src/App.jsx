// @flow

import React from 'react'
import Websocket from 'react-websocket'
import {compose} from 'ramda'

import {Siths_, Planet_, List_} from './state/connectors'
import Layout, {Sith, Header, Body, PlanetIndicator, ScrollableList, List, Item, ScrollButtons, ScrollButton} from './ui/'

import {name, homeworld} from './ui/utils'

const PLANET_URL = 'ws://localhost:4000'

const SITH_URL = 'http://localhost:3000/dark-jedis/'
import sithActions from './state/actions/siths'

const App = () => <Layout>

  <Planet_>{({planet, planetReceived, dispatch}) => <div>
    <Websocket url={PLANET_URL} onMessage={(serverPlanet: string) => compose(planetReceived, JSON.parse)(serverPlanet)}/>

    <Header>
      <PlanetIndicator planet={planet} />
    </Header>

    <Body>
      <Siths_>{({siths}) =>
        <List_>{({list, scrolled}) =>
          <ScrollableList>

            <List>
              <Item><Sith name={name(siths[list[0]]) || ''} homeworld={homeworld(siths[list[0]]) || {}} /></Item>
              <Item><Sith name={name(siths[list[1]]) || ''} homeworld={homeworld(siths[list[1]]) || {}} /></Item>
              <Item><Sith name={name(siths[list[2]]) || ''} homeworld={homeworld(siths[list[2]]) || {}} /></Item>
              <Item><Sith name={name(siths[list[3]]) || ''} homeworld={homeworld(siths[list[3]]) || {}} /></Item>
              <Item><Sith name={name(siths[list[4]]) || ''} homeworld={homeworld(siths[list[4]]) || {}} /></Item>
            </List>

            <ScrollButtons>
              <ScrollButton direction={'UP'} onClick={scrolled} />
              <ScrollButton direction={'DOWN'} onClick={scrolled} />
            </ScrollButtons>

          </ScrollableList>}
        </List_>
      }</Siths_>
    </Body>

    <button onClick={() =>
      fetch(`${SITH_URL}${3616}`)
        .then((response) => response.json())
        .then((sith) => dispatch(sithActions.sithReceived(sith)))}>Start</button>
  </div>}</Planet_>

</Layout>

export default App
