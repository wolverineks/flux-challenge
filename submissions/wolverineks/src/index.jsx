// @flow

import React from 'react'
import ReactDOM from 'react-dom'
import {Provider} from 'react-redux'
import {attachPixie} from 'redux-pixies'

import createStore from './state/store'
import App from './App.jsx'
import appPixie from './async'

const store = createStore({})

attachPixie(store, appPixie)

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
