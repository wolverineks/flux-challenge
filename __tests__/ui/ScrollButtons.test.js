// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import {ScrollButtons, ScrollButton, classNames} from '../../src/ui/'

describe('ScrollButtons', () => {
  describe('When has buttons', () => {
    test('ScrollButtons', () => {
      const className = classNames.scrollButtons
      const component = renderer.create(<ScrollButtons className={className}>
        <ScrollButton direction={'UP'} onClick={() => {}} />
        <ScrollButton direction={'DOWN'} onClick={() => {}} />
      </ScrollButtons>)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
