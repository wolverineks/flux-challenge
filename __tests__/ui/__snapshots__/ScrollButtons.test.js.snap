// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollButtons When has buttons ScrollButtons 1`] = `
<div
  className="css-scroll-buttons"
>
  <button
    className="css-button-up"
    onClick={[Function]}
  />
  <button
    className="css-button-down"
    onClick={[Function]}
  />
</div>
`;
